core/arch/riscv64/memfuncs.asm

```diff
@@ -44,30 +44,130 @@ START_FILE
  */
         DECLARE_FUNC(memcpy)
 GLOBAL_LABEL(memcpy:)
-/* TODO i#3544: Naive version, optimize it. */
-        mv       t1, ARG1
-        beqz     ARG3, 2f
-1:      lbu      t2, 0(ARG2)
-        addi     ARG2, ARG2, 1
-        sb       t2, 0(t1)
-        addi     t1, t1, 1
-        addi     ARG3, ARG3, -1
-        bnez     ARG3, 1b
-2:      ret
+        addi     a5, x0, 32
```

Comment:
I'm not very familiar with RSIC-V so please bear with me:

This is just putting 32 into a temp reg; should it use t3-t6 instead of a5?

Reply:
Right. I have modified this by using t6.